packages/headless/src/controllers/commerce/core/facets/numeric/headless-commerce-numeric-facet.ts

```diff
@@ -28,6 +59,14 @@ export type NumericFacet = CoreCommerceFacet<
   NumericRangeRequest,
   NumericFacetValue
 > & {
+  /**
+   * Replaces the current range values with the specified ones.
+   * @param ranges - The new ranges to set.
+   */
+  setRanges: (ranges: NumericFacetValue[]) => void;
```

Comment:
I would have assumed this would be capped to a length of two, to set the min and the max of the currently selected range.

Why is this an array?

Reply:
Initially, I had made the function accept a single range, but I decided to go with an array because this will allow developers to implement numeric facets into which the user can enter arbitrary ranges (e.g., by using numeric inputs).

When the user adds a new range in this fashion, the facet would call the setRanges method and pass the current ranges + the new range. We can validate that the user-entered range is valid by comparing it to the facet's domain.